JavaScript Capstone Project

HFG43 opened the pull request:
Hi!! Thanks for your previous feedback! ✔🙌
As requested, we have followed the changes required and suggestions in the README file. 
we have changed the following:

1. Update README with installation data
2. Include two more scenarios to comments count test
3. Improve show cards display as requested

Hi there! @Kidd254 and @HFG43 are sending this Pull Request for you to review our JavaScript Capstone project.
Thanks in advance for your feedback! 😎🙌🙌
For this project, we have worked and done the following:
We have developed a dynamic search engine for the app, that let us GET the info from the TVmaze and Microverse Involvement APIs, and look for different genres and topics on demand.
•	  Use Kanban as a project management tool, and complement with daily status meetings ✔
•	  Work doing pair programming ✔
•	  Use GitFlow ✔
•	  Review each other's code ✔
•	  Follow wireframe as app skeleton design ✔
•	  Worked using ES6 syntax and modules ✔
•	  Solve merge conflicts using git ✔
•	  Add display TV shows (cards) dynamically for Home Page and pop up ✔
•	  Add counter for all elements in DOM displayed, likes, and comments. each of the with the functionality to update upon instant changes ✔
•	  Run tests using JEST as requested for all elements counter, and counter of comments ✔
•	  Record a video to explain app functionalities and show part of our code ✔
•	  Review Linters ✔
•	  Use Webpack ✔
•	  Deploy the project using GitHub pages ✔

HFG43 committed:
Configure Webpack

Kidd254 committed:
add es linter configuration files

Kidd254 committed:
configure es linter

HFG43 committed:
Run Eslint & Stylelint commands for instalation

HFG43 committed:
Add index HTML for basic structure

HFG43 committed:
Add Footer Section to HTML & Include MIT LICENSE

Kidd254 committed:
configure for images

Kidd254 committed:
import the css file

Kidd254 committed:
add an icon for the header

Kidd254 committed:
add basic header in the header section

Kidd254 committed:
update the relevant dependencies affected by change

HFG43 committed:
Update webpack.config.js to resolve output issue

Kidd254 committed:
add a fonts directory

Kidd254 committed:
update webpack.config.js file to accomodate fonts

Kidd254 committed:
add a directory to img file

Kidd254 committed:
add font styles

Kidd254 committed:
add an icon for the header

Kidd254 committed:
update the json file to outomatically load live server

Kidd254 committed:
update the dist folder

Kidd254 committed:
configure webpack to display images

Kidd254 committed:
update the css file for perfect alignment of the header and icon

Kidd254 committed:
update the dist folder and the relevant dependencies

HFG43 committed:
Add Copyright

Kidd254 committed:
Merge pull request #25 from HFG43/Header

Adding the Header and the Webpack Assets

HFG43 committed:
merge footer & header

HFG43 committed:
Merge pull request #22 from HFG43/Footer

Add Footer

HFG43 committed:
Solve Merge Conflicts

Header, Assets and Footer in place

HFG43 committed:
Merge branch 'feature/Day_Zero' into dev

HFG43 committed:
Add Home Page Skeleton

Index Getting API Elements
Search Function to have a dynamic search in the app
Move License to root Menu

HFG43 committed:
Place SearchInput holder in place

HFG43 committed:
Add Display Responsive Style

HFG43 committed:
Add conditionale Image Display

Ensure all elements displayed have image != null from API input

HFG43 committed:
Arrange Like Button Position in TV Show Card

HFG43 committed:
Add Validation at search Input to improve UX

HFG43 committed:
Correct Linters - Style Input Validation

Kidd254 committed:
add popup card dynamicaly

Kidd254 committed:
add styling to dynamic content

Kidd254 committed:
add tags for dynamic display of the popup

Kidd254 committed:
update the dist folder files

Kidd254 committed:
modify js and css files

Kidd254 committed:
implement popup

Kidd254 committed:
update dist files

HFG43 committed:
Add genre runtime type and language values to newShow

HFG43 committed:
Merge pull request #26 from HFG43/feature/HomePage

Feature/home page

Kidd254 committed:
add the popup function to separate file

Kidd254 committed:
resolve

Kidd254 committed:
correct popup function

Kidd254 committed:
add details to popup

Kidd254 committed:
corrent es lint errors

Kidd254 committed:
modify the aspect ratio of the pop up image

Kidd254 committed:
Merge pull request #27 from HFG43/popUp-details

Pop up details

Kidd254 committed:
break down the code into modules

Kidd254 committed:
break down a massive code

Kidd254 committed:
Merge branch 'popUp-details' into pop-comments

Kidd254 committed:
add popup comment

Kidd254 committed:
update the dist files

Kidd254 committed:
add tags for the popup comment

Kidd254 committed:
update dist files

HFG43 committed:
Test Setting AppID

Kidd254 committed:
Merge pull request #28 from HFG43/pop-comments

Pop comments

Kidd254 committed:
add form element skeleton dynamicaly

Kidd254 committed:
add styling to the form elements

Kidd254 committed:
update dist files

Kidd254 committed:
add an event listener to add comment button

Kidd254 committed:
add function that will loop through all comments and display them

Kidd254 committed:
modify the popup function to display comments

Kidd254 committed:
add involvement api

Kidd254 committed:
update dist files

HFG43 committed:
Set Get function in API

HFG43 committed:
Add postLike function at likes.js

Function wroking ok; response === created

HFG43 committed:
Add function to get Likes

Get Array working ok!

Kidd254 committed:
modify the create comment function

Kidd254 committed:
update dist files

HFG43 committed:
Add likeCounter element to show card

HFG43 committed:
Add Likes counter for update and onload

HFG43 committed:
Finish Add new like to show functions

It posts new like upon click to the tv show

HFG43 committed:
Review Linters

Kidd254 committed:
clear the input fields in the popup after clicking comment button

Kidd254 committed:
clear the input fields in the popup after clicking comment button

Kidd254 committed:
modify the comment button to display comments

Kidd254 committed:
update the dist files

Kidd254 committed:
add callback to add comments after the comment button is clicked

Kidd254 committed:
add a function to prevent adding empty comment or default

Kidd254 committed:
update dist files

Kidd254 committed:
prevent default

HFG43 committed:
Merge branch 'dev' of https://github.com/HFG43/JS_Capstone into feature/like

HFG43 committed:
Reverted Merge

I mistakenly had merge it the other way round (dev into like feature)

HFG43 committed:
Merge pull request #30 from HFG43/feature/like

Feature/like

HFG43 committed:
Original Index.js and likes.js approved in display_likes branch

Kidd254 committed:
update comments.js

HFG43 committed:
Add default load category

Kidd254 committed:
merge

Kidd254 committed:
Merge pull request #29 from HFG43/add-newComment

Add new comment

HFG43 committed:
Modify homePage display position

Readapt to Wireframe, improve RWD & UX

HFG43 committed:
Correct Linters

Replace function to load default shows

Kidd254 committed:
add a function to count the number of comments

Kidd254 committed:
update dist files

HFG43 committed:
Add functions to count and display all TV show items

Kidd254 committed:
display number of comments

Kidd254 committed:
fix an error

Kidd254 committed:
configure jest

Kidd254 committed:
configure jest

Kidd254 committed:
add a test for comments counter

Kidd254 committed:
configure jest by adding to package Json

Kidd254 committed:
update dist files

Kidd254 committed:
change

HFG43 committed:
Merge pull request #34 from HFG43/all_items_counter

All Items Counter

Kidd254 committed:
fix a comments bug

Kidd254 committed:
submit after all input fields are added

Kidd254 committed:
fix linters

Kidd254 committed:
Merge pull request #33 from HFG43/commentsCounter

Comments counter

HFG43 committed:
Merge Display Likes to Dev & fix linters errors

HFG43 committed:
Merge pull request #32 from HFG43/Solve_Display_Likes

Display_Likes (fix bug)

HFG43 committed:
Replace like counters and it's fon size

Add padding for better view in TV Shows card using mobile

HFG43 committed:
Install Jest dependencies

HFG43 committed:
Merge pull request #37 from HFG43/Jest

Install Jest dependencies

Kidd254 committed:
add changes

Kidd254 committed:
merge

HFG43 committed:
Update README file

Kidd254 committed:
add tests for the comments counter function

HFG43 committed:
Add GH Pages Deployment Link

Kidd254 committed:
fix linters

HFG43 committed:
Add test Functions for Count all elements in Home Page

HFG43 committed:
Merge branch 'dev' into all-elements-count-test

HFG43 committed:
Review Linters

Kidd254 committed:
update dist files

HFG43 committed:
Merge branch 'all-elements-count-test' of https://github.com/HFG43/JS_Capstone into all-elements-count-test

Kidd254 committed:
merge conflicts

HFG43 committed:
Merge pull request #39 from HFG43/all-elements-count-test

Add test Functions for Count all elements in Home Page

Kidd254 committed:
resolve merge conflict

Kidd254 committed:
Merge pull request #35 from HFG43/test-commentCounter

Test comment counter

HFG43 committed:
Correct Linters errors

HFG43 committed:
Run "npm run build" to update GH Pages Development

HFG43 committed:
Merge branch 'master' into dev

HFG43 committed:
Linter errors

HFG43 committed:
Merge branch 'dev' of https://github.com/HFG43/JS_Capstone into dev

danmainah commented on src/popUp.js:
```diff
@@ -0,0 +1,111 @@
+          <button type="submit" class="add-comment" id="comment-btn">Comment</button>
+        </form>
+      </div>
+     </div>
```
- [ ] @HFG43 and @Kidd254 Great work adding the popup window, however someone cannot view the whole content in the popup without scrolling, kindly consider ensuring all content is visible like shown in the initial mockup, check the comparison below:


Mock-up Design          |  Your design     
:-------------------------:|:-------------------------:
![image](https://github.com/HFG43/JS_Capstone/assets/61408860/fec64ba5-1409-42a2-b332-b6b287b75faf) | ![image](https://github.com/HFG43/JS_Capstone/assets/61408860/fad13391-5ebf-46d9-abbf-34bfe8d987d2)

danmainah commented on src/commentsCounter.test.js:
```diff
@@ -0,0 +1,26 @@
+import commentsCounter from './commentsCounter.js';
+
+describe('commentsCounter', () => {
+  test('displays correct comments count', () => {
+    // Mock the necessary elements
+    document.body.innerHTML = `
+      <div>
+        <h3 id="comments-header"></h3>
+        <ul>
+          <li class="list-items"></li>
+          <li class="list-items"></li>
+          <li class="list-items"></li>
+        </ul>
+      </div>
+    `;
+
+    // Call the commentsCounter function
+    commentsCounter();
+
+    // Get the updated comments count
+    const commentsCount = document.getElementById('comments-header').innerText;
+
+    // Verify that the comments count is displayed correctly
+    expect(commentsCount).toBe('Comments (3)');
+  });
```
- [ ] @team great work adding tests for the comments counter, however, you have tested only one scenario, kindly add tests for more scenarios eg when there are no comments, or different numbers of comments.

danmainah commented on README.md:
```diff
@@ -0,0 +1,188 @@
+
+To run the project, execute the following command:
+
+- In the command line:
```
- [ ] [optional] It would have been nice if you added the command `npm install` that installs dependencies in package.json to make the app work.

danmainah reviewed (changes requested):
Hi @HFG43 and @Kidd254 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Ensured all the functions are working so the user can like to add comments and search for a tv show
✔️  Good design on the page that is responsive giving better user experience in mobile and desktop view.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

HFG43 committed:
Update README

Include install npm command in "installation"

HFG43 committed:
Improve pop up image display

Change size in image and increase pop up size to ensure is better seen by users

HFG43 committed:
Add test scenarios

Add zero and 1 counter scenario for comments counter testing

HFG43 committed:
Run build for GH Page deployment

abdulhamiid reviewed (approved):
# Approved 🟢
Hi @HFG43 👋

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
